deploy: merge upstream Spack v0.19.0

heerener opened the pull request.

haampie committed:
tau: checksum (#33873)

alalazo committed:
Fix a bug/typo in a config_values.py fixture (#33886)

haampie committed:
Fix incorrect timer (#33900)

Revamp the timer so we always have a designated begin and end.

Fix a bug where the phase timer was stopped before the phase started,
resulting in incorrect timing reports in timers.json.

haampie committed:
Fixup: start the timer before the phase (#33917)

matz-e reviewed (changes requested):
Fix the build failures, please.

michaelkuhn committed:
docs: fix typo (#33926)

heerener committed:
Pin stat at previous version in the hope that it works

Try removing stat. The version pin didn't help.

NadirRoGue committed:
Adds Brayns 3.0.0 and new py-brayns package (#1767)

* Adds Brayns 3.0.0 and new py-brayns package

* Fix style

* Update viz environment

* Address review

heerener committed:
Revert "Add binutils module to nvhpc in compilers.yaml (#1418)"

This reverts commit 9431135148aa3477fd7178740664f308c057dd44.

heerener committed:
Add comments so we can find back what we disabled

In our desperate attempts to get the pipeline to work after merging
upstream

heerener committed:
Term blacklist was replaced with exclude

haampie committed:
docs: fix typo in multiple build systems

iomaganaris committed:
Update neurodamus-{hippocampus,neocortex,common} (#1768)

* Use common commit for fixing mcr4 clearing
* Update neurodamus models with new versions with fixes of clearing mcran4 objects

NadirRoGue committed:
Adds missing module to viz environment (#1769)

heerener committed:
Disable some packages which break

genric committed:
bump bbp-workflow, add bbp-workflow-cli (#1766)

heerener committed:
Deduplicate dependency list

matz-e committed:
neuroNdamus: fix packages

matz-e committed:
deploy: fix intel setup with our gcc

heerener committed:
sim-model: cleanup

heerener committed:
Set cuda_arch and re-enable science packages

heerener committed:
py-cmake-format: fix up

heerener committed:
py-atlas-building-tools: disable

It needs py-numba but no versions are available

heerener committed:
py-cmake-format: add sha for version 0.6.11

mgeplf committed:
atlas-densities v0.1.2 -> v0.1.3 (#1770)

alalazo committed:
Improve error message for requirements (#33988)

refers #33985

heerener committed:
py-cmake-format: re-enable

greenc-FNAL committed:
PackageBase should not define builder legacy attributes (#33942)

* Add a regression test for 33928

* PackageBase should not set `(build|install)_time_test_callbacks`

* Fix audits by preserving the current semantic

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

heerener committed:
py-bluepyefe: remove BPE2 version

heerener committed:
Don't try to add args for dep which is not in spec

haampie committed:
uninstall: fix accidental cubic complexity (#34005)

* uninstall: fix accidental cubic complexity

Currently spack uninstall runs in worst case cubic time complexity
thanks to traversal during traversal during traversal while collecting
the specs to be uninstalled.

Also brings down the number of error messages printed to something
linear in the amount of matching specs instead of quadratic.

heerener committed:
Less parallelization to avoid possible locking issues

scottwittenburg committed:
ci: restore ability to reproduce gitlab job failures (#33953)

heerener committed:
Pin py-cmake-format to the last-known working version

heerener committed:
highfive: add 2.6.0, 2.6.1 and 2.6.2 versions

heerener committed:
py-python-keycloak: needs py-poetry to build

heerener committed:
py-querystring-parser: depends on py-setuptools to build

heerener committed:
py-pyld: depends on py-setuptools to build

matz-e committed:
{py-,}mvdtool: fixeth

heerener committed:
py-opencv-python: version bump

heerener committed:
py-brayns: needs py-setuptools to install

heerener committed:
py-opencv-python: fix pypi package name

matz-e commented on bluebrain/repo-patches/packages/py-opencv-python/package.py:
```diff
@@ -10,9 +10,9 @@
 class PyOpencvPython(PythonPackage):
     """Pre-built CPU-only OpenCV packages for Python."""
 
-    homepage = "https://pypi.org/project/opencv-python"
```
Homepage should still be there :)

heerener committed:
deploy: Uninstall intel compilers before getting to slurmspack

genric committed:
py-bbp-workflow-cli: bump (#1772)

heerener committed:
spatial-index: requires boost+filesystem

heerener committed:
boost: specify +filesystem properly

heerener committed:
augment: bugfix in command

heerener committed:
brion: purge older versions

heerener committed:
arrow: add build flags and purge old versions

matz-e committed:
hpctoolkit: enable MPI (#1773)

heerener committed:
py-atlannot: version bump

pramodk committed:
neurodamus-model: Fix bug in build_neurodamus.sh (#1774)

* list of additional mod files are read from file and those
    contain trailing newline char
  * comparison done between two list was returning undesired
    list of mod files due to leading newline char

jorblancoa committed:
[core]neuron: updates and fixes

CoreNEURON: reduce memory usage in report initialization (#1776)

Recipe changes to combine coreneuron into NEURON (#1777)

* Bring coreneuron recipe options into neuron for coreneuron merge
* remove dependency with coreneuron when neuron>=9.0 or develop
* coreneuron variant is on by default in neuron
* coreneuron mechanism library should always have different name than
   libcorenrnmech.so as it's now part of NEURON installation
* do not add arch specific opt flags with +debug
* Automatically bring coreneuron build_type=Debug when neuron+debug
  is used. This will simplify logic in blueconfig CI where we don't
  have to explicitly specify coreneuron dependency (especially as we
  have merged neuron and coreneuron)
* Find "right" nrnivmodl-core binary for neurodamus building

See relevant PR: https://github.com/neuronsimulator/nrn/pull/2055

Deploy NEURON's upcoming 9.0a release (#1781)

* Updated model-neocortex packages with latest 1.10 tag
* add nmodl 0.5.a
* remove coreneuron packages

Co-authored-by: Ioannis Magkanaris <iomagkanaris@gmail.com>
Co-authored-by: Matthias Wolf <matthias.wolf@epfl.ch>

Update NEURON version (#1792)

- Create NEURON version 9.0.a2 that includes bug fixes

[Core]NEURON 8.2.2 (#1802)

coreneuron: so long, and thanks for the fish.

Coreneuron has been folded into neuRon and should not be referred to as
a standalone package any more.

Add py-sympy dependency at neuron package level

newer spack doesn't seem to propogate python package
dependencies (as PYTHONPATH) during build phase.

neuron: omitted link flag for Caliper

alalazo committed:
Use a module-like object to propagate changes in the MRO, when setting build env (#34059)

This fixes an issue introduced in #32340, which changed the semantics of the "module"
object passed to the "setup_dependent_package" callback.

heerener committed:
Coreneuron and neuron: require %intel OR %nvhpc

heerener committed:
neuron: %nvhpc is taken care of by packages.yaml

heerener committed:
py-atlannot: v0.1.3 now on pypi

matz-e committed:
highfive: fudge more

iomaganaris committed:
py-neurodamus: update to 2.12.10 (#1779)

heerener committed:
asciitoh5: remove

heerener committed:
.gitlab-ci.yaml: update DEPLOYMENT_DATE

matz-e committed:
neuRonDamus: fix recipes

heerener committed:
petsc: fix forced scalapack

1uc committed:
ultraliser: update (#1782)

heerener committed:
petsc: better fix for forced scalapack

This reverts commit ab03d912f13afdc14157c79207890c0411ec3204.

1uc committed:
hdf5: Patch pagebuffer check in `H5Fopen`. (#1778)

The pagebuffer should be compatible with pHDF5, as long as
one doesn't use MPIO. This commit allows one request patching
HDF5 to weaken the check. The variant is called `+page_buffer_patch`.

This is only supported for HDF5 `v1.12.1`. It's used in the context
of reading morphologies from single-file HDF5 morphology containers.
One application that requires this is TD.

heerener committed:
py-basalt: pin dependency & remove from deployment

1uc committed:
morpho-kit: now with paged HDF5. (#1783)

eleftherioszisis committed:
brainbuilder: add version 0.18.1 (#1785)

matz-e committed:
deploy: fix a bunch of compiler stuff

matz-e committed:
brion: fix boost dependency

matz-e committed:
deploy: remove more nvhpc customization

matz-e committed:
nmodl: unbreak recipe

matz-e committed:
neuron: calipre is a link dependency, remove customißation

heerener committed:
deploy: cleanup and fixes

heerener committed:
The Purge 73: Patches

heerener committed:
llvm, nvhpc, clang-tools: bump

heerener committed:
mxnet: patch

heerener committed:
packages: get rid of nvhpc

matz-e committed:
node-js: version bump

matz-e committed:
libuv-julia: fix dependencies

matz-e committed:
julia: revert customization, remove external automake instead

matz-e committed:
m4: remove external

matz-e committed:
libuv-julia: reintroduce dependency

wyphan committed:
nvhpc: add v22.11 (#34410)

matz-e committed:
py-sh: changes have been pushed upstream

alalazo committed:
scons: fix signature for `install_args` (#34481)

heerener committed:
cuda: new version and all the fun that comes with it

heerener committed:
bazel: bump

heerener committed:
deploy: No GPU variants for now

heerener committed:
py-tensorflow: remove

py-tensorflow-addons: package rewrite

matz-e committed:
morphio: turn off warnings

matz-e committed:
py-submitit: upstreamed

matz-e committed:
python-dev: weed out stuff

matz-e committed:
brion: attempt some fixes

matz-e committed:
brayns: patch for GCC 12

alalazo committed:
Forward lookup of the "run_tests" attribute (#34531)

fixes #34518

Fix an issue due to the MRO chain of the package wrapper
during build. Before this PR we were always returning
False when the builder object was created before the
run_tests method was monkey patched.

heerener committed:
llvm: build 14.0.6 to make py-libclang happy

alalazo committed:
Propagate exceptions from Spack python console (#34547)

fixes #34489

Customize sys.excepthook to raise SystemExit when
any unhandled exception reaches the hook.

heerener committed:
Trigger rebuild

heerener committed:
julia: pin to 1.6

matz-e commented on bluebrain/deployment/environments/externals.yaml:
```diff
@@ -119,7 +117,7 @@ spack:
     - intel-oneapi-mkl@2021.4.0
     - ispc
-    - julia
+    - julia@1.6
```
```suggestion
    - julia@1.6  # As requested by Polina S.
```

matz-e committed:
Update bluebrain/deployment/environments/externals.yaml

heerener committed:
julia: disable for now

heerener committed:
hpc: disable GPU for now

eleftherioszisis committed:
brainbuilder: update version to 0.18.2 (#1794)

eugeneswalker committed:
e4s: disable mac stack due to binary relocation issue#32571 (#34560)

iomaganaris committed:
neurodamus-model: Update common submodule with changes related to vascouplingB (#1796)

eleftherioszisis committed:
brainbuilder: update version to 0.18.3 (#1797)

Fixes for numpy 1.24 deprecation errors.

mgeplf committed:
py-nexusforge: bump to 0.8.0, remove 0.6.3 and dependencies (#1793)

* remove py-rdflib-jsonld, unshackle pyshacl
* unpin py-prettytable

Co-authored-by: Matthias Wolf <matthias.wolf@epfl.ch>

matz-e committed:
spatial-index: purge older versions.

matz-e committed:
regiodesics: pull archived software from deployment

matz-e committed:
py-bluepyemodel: purge ancient deps

matz-e committed:
py-atlas-direction-vectors: purge: F NUMBA and its LLVM 11 dependency

matz-e committed:
py-bleupyopt: bump

iomaganaris committed:
neurodamus-py: update version

* Update `neurodamus-<model>`s modules' naming to reflect changes in `common` as well

tgamblin committed:
docs: remove monitors and analyzers (#34358)

These experimental features were removed in #31130, but the docs were not.

- [x] remove the `spack monitor` and `spack analyze` docs

haampie committed:
Fix combine_phase_logs text encoding issues (#34657)

Avoid text decoding and encoding when combining log files, instead
combine in binary mode.

Also do a buffered copy which is sometimes faster for large log files.

1uc committed:
SpatialIndex: 1.2.0 (#1786)

* SpatialIndex 1.2.0

* Add also 1.2.1 version

* SpatialIndex and 'py-libsonata@0.1.15` are incompatible.

Co-authored-by: Antonio Bellotta <antonio.bellotta@epfl.ch>

tristan0x committed:
Steps: 2023 (#1800)

* Backport py-build and py-pep517 from v0.19.0

* steps: requires py-build starting STEPS 5

* steps: ~native on MacOS by default

heerener committed:
bazel: fix 5.1.1 build

heerener committed:
ultraliser: fix github url and bump version

1uc committed:
morphio: build without warnings, to avoid `-Werror`. (#1718)

Build MorphIO without warnings, to avoid `-Werror`; and modernize formatting.

heerener committed:
steps: remove py-unittest2 dependency

Fortunately it is not used, as it is broken on python3.10

heerener committed:
audit: correct patch URLs

heerener committed:
py-bluepyemodel: too many uppercases in tag

heerener committed:
touchdetector: fix KeyError

heerener committed:
py-pyshacl: add py-poetry dependency

mgeplf committed:
atlas-direction-vectors: bump to v0.1.4, drop v0.1.2 (#1801)

* bump atlas-direction-vectors to v0.1.4, drop v0.1.2

Co-authored-by: Matthias Wolf <matthias.wolf@epfl.ch>

heerener committed:
nmodl: bump fmt dependency

blue42u committed:
FileCache: Delete the new cache file on exception (#34623)

The code in FileCache for write_transaction attempts to delete the temporary file when an exception occurs under the context by calling shutil.rmtree. However, rmtree only operates on directories while the rest of FileCache uses normal files. This causes an empty file to be written to the cache key when unneeded.

Use os.remove instead which operates on normal files.

Co-authored-by: Harmen Stoppels <harmenstoppels@gmail.com>

heerener committed:
model-neocortex: cleanup package

mgeplf committed:
regiodesics: bump from 0.1.2 to 0.1.3 (#1804)

* allows more modern boost

matz-e committed:
deploy: various fixes

deploy: approximate bbp/develop better

deploy: remove gene-expression-volume, py-atlalign

OK'd by NSE

deploy: tune requirements

deploy: cleaner Intel configuration

heerener committed:
deploy: concretizer:unify:false

eugeneswalker committed:
protected runner jobs at uo: trust e4s-uo signing key (#34828)

matz-e committed:
nmodl: attempt to specify dependencies "better"

heerener committed:
py-torch: patches and dependencies

pramodk committed:
trilinos: try to build newer @13.0.1

ggouaillardet committed:
fix libtool filter for Fujitsu compilers (#34916)

Since legit regular expressions are matched, do not pass string=True
to the filter

Refs #34897

tristan0x committed:
Omega-h: update deployment spec / new variant (#1808)

* add variant to only enable Kokkos
* default deployment on BB5 relies only on Kokkos, not Trilinos
Conflicts:
	bluebrain/deployment/environments/applications_libraries.yaml

haampie committed:
spack mirror create takes local paths only (#34992)

spack mirror create cannot work with urls, so it shouldn't promote local
paths to file urls.

matz-e committed:
nvhpc: deploy new version

matz-e committed:
py-torch: various attempts at fixes

py-torch, cuda: more appeasement

py-torch: more "customizations" to force things along

py-torch: pin cudNN

py-torch: compiler-dance

py-torch: compiler dance, part dieux

py-torch: more variant confusion 🙈

haampie committed:
disable flake warnings for unused imports of types

climbfuji committed:
Bug fix for duplicate rpath errors on macOS when creating build caches (#34375)

haampie committed:
no python 3.6 anymore on ubuntu 22.04 in relevant setup action?

haampie committed:
buildcache create: make "file exists" less verbose (#35019)

Currently we print 3 lines of "file exist" warning per tarball, this is
a bit excessive. Instead, it can be a simple single-line tty.warn
message.

matz-e committed:
py-atlinter: truncate versions, add new one.

matz-e committed:
py-shortuuid: sync with spack/spack@develo

matz-e committed:
py-flatten-dict: switch to poetry for builds

heerener committed:
py-frozendict: now exists in builtin

heerener committed:
incorporate upstreamed changes

heerener committed:
py-trimesh: fix dependency name

matz-e committed:
deploy: more MPI in NSE

heerener committed:
py-gunicorn: already upstreamed

matz-e committed:
deploy: pin ipyparallel

matz-e committed:
deploy: generate more brion modules.

heerener committed:
py-nexusforge: bump dependencies to get rid of python2

heerener committed:
py-ipyparallel: version bump

heerener committed:
py-deap: newer version

matz-e committed:
py-flatten-dict: sync with upstreamed version.

matz-e committed:
py-plotly: bump version and dependencies.

matz-e committed:
py-vascpy: fix build dependencies

matz-e committed:
py-voxcell: fix dependencies

matz-e committed:
py-vascpy: typo

heerener committed:
py-lazy: upstreamed version bump

heerener committed:
py-pyspark: patch got upstreamed

py-py4j: patch got upstreamed

matz-e committed:
py-bluepyefe: fix dependencies

alalazo committed:
Add a `maintainers` directive (#35083)

fixes #34879

This commit adds a new maintainer directive,
which by default extend the list of maintainers
for a given package.

The directive is backward compatible with the current
practice of having a "maintainers" list declared at
the class level.

matz-e committed:
py-atlannot: truncate versions, fix dependencies

matz-e committed:
bríon: actually extends Python.

matz-e committed:
py-pyarrow: pass options through environment

matz-e committed:
deploy: rebuild intel

deploy: purge intel stuff earlier
deploy: don't uninstall intel stuff any longer

heerener committed:
py-nptyping, py-typish: backport upstreamed changes

heerener committed:
py-ipyparallel: backport upstreamed changes

heerener committed:
py-nexusforge: backport upstreaming with dependencies

heerener committed:
py-hjson: fix typo in version

matz-e committed:
deploy: re-enable external Julia

heerener committed:
py-mariadb: upstreamed

matz-e committed:
llvm: make patch optional

pramodk committed:
py-pebble: add newer versions (for bluepyopt)

heerener committed:
julia: fix for KeyError

heerener committed:
py-jinja2-cli: upstreamed

py-matplotlib-scalebar: upstreamed
py-munkres: upstreamed
py-equation: upstreamed

matz-e committed:
core: fix path handling in prefix inspections

At least with ZSH, prefix inspections containing `./bin` result in a
`$PREFIX/./bin` and result in strange `$PATH` handling.

I.e., `module load git` will prepend `/path/to/git/./bin`, `which git`
will find the right executable, but `git --version` will print the
system one. Normalize the relative path to investigate to avoid this
behavior.

See also spack/spack#31867.

heerener committed:
amdlibflame: update from upstream/develop

amdlibflame: needs modern make

See https://github.com/flame/libflame/issues/55

heerener committed:
gmsh: re-enable

ultraliser: re-enable

heerener committed:
py-pyshacl: fix dependency version

matz-e committed:
regiodesics: fix compilation

matz-e committed:
deploy: farewell, gcc@11.

matz-e committed:
deploy: remove old Intel compiler.

heerener committed:
py-netpyne: upstreamed with dependencies

heerener committed:
py-igor: upstreamed

matz-e committed:
Merge remote-tracking branch 'bbp/develop' into up-the-stream

matz-e committed:
Fix path handling in prefix inspections (#35318)

At least with ZSH, prefix inspections containing `./bin` result in a
`$PREFIX/./bin` and result in strange `$PATH` handling.

I.e., `module load git` will prepend `/path/to/git/./bin`, `which git`
will find the right executable, but `git --version` will print the
system one. Normalize the relative path to avoid this behavior.

See also spack/spack#31867.

haampie committed:
Revert "Hotfix/namespace (#33870)"

This reverts commit a86911246a8c5571ca210ff25fc990b8eb297a5c.

haampie committed:
Set version to 0.19.1.dev0

matz-e committed:
deploy: fix compilers, neuRon, patches

haampie committed:
Set version to 0.19.1

scottwittenburg committed:
bindist: use append as a method, not assignable attr (#35379)

heerener committed:
py-poetry-plugin-export: get latest upstream

to get rid of circular dependency

heerener committed:
cvise: upstreamed

heerener committed:
py-luigi-tools: missing py-setuptools-scm

heerener committed:
py-cwl-registry: update dependencies

heerener committed:
py-cwl-luigi: update dependencies

matz-e committed:
deploy: remove archived software as agreed 2023-02-09

matz-e committed:
deploy: remove garbage

matz-e committed:
deploy: remove archived, outdated packages

OK from the Viz team and me for the HPC packages

matz-e committed:
core: workaround for "fixing" disappearing packages.

When we upstream packages, we delete our copy and update the builtin
version. But the database may still refer to the copy, so "bend" things
a little to avoid getting Python exceptions.

matz-e committed:
deploy: remove more packages

matz-e committed:
touchdetector: unbreak

matz-e committed:
deploy: remove outdated tox, generate pytest module

matz-e committed:
deploy: add GCC gunk to modern 1api™ compilers

matz-e committed:
deploy: more cleanup

matz-e committed:
deploy: another attempt to fix 1api™ compilers.

matz-e committed:
deploy: undo module additions

heerener committed:
py-cmake-format, py-pynrrd: upstreamed patches

heerener committed:
py-mlflow: upstreamed with dependencies

matz-e committed:
Merge remote-tracking branch 'origin/releases/v0.19' into up-the-stream

matz-e committed:
deploy: update pipeline helper script

matz-e committed:
deployment: forget about hpc* tools that duplicate Pythons

@pramodk FYI.

pramodk commented:
@matz-e : about the a488d7cbdd1d3e78c997f5b759acc727aafa22ee - we will need hpctoolk one way or another. So soon we will need to find a solution to this.

matz-e committed:
deploy: pin stuff to skylake

matz-e committed:
deploy: there shall be no MPI besides hpe-mpi

psakievich committed:
Allow users to specify root env dir (#32836)

* Allow users to specify root env dir

Environments managed by spack have some advantages over anonymous Environments
but they are tucked away inside spack's directory tree. This PR gives
users the ability to specify where the environments should live.

See #32823

This is also taken as an opportunity to ensure that all references are to "managed environments",
rather than "named environments". Prior to this PR some references to the latter persisted.

Co-authored-by: Tom Scogland <scogland1@llnl.gov>
Co-authored-by: Tamara Dahlgren <35777542+tldahlgren@users.noreply.github.com>
Co-authored-by: Gregory Becker <becker33@llnl.gov>
Conflicts:
	lib/spack/spack/test/cmd/env.py
	lib/spack/spack/test/env.py

matz-e committed:
zee: remove abandoned package with +oce dependencies

matz-e committed:
ospray: seems to work again with target=skylake

matz-e committed:
deploy: be less stringent on reportinglib

matz-e committed:
nvhpc: backport 23.1

matz-e committed:
deploy: another rebuild from scratch

matz-e committed:
brayns: update patch sha

matz-e committed:
deploy: configure environments to be stored in the user cache path

matz-e committed:
Merge remote-tracking branch 'bbp/develop' into up-the-stream

matz-e committed:
deploy: remove another environment customization remnant.

matz-e committed:
steps: purge ancient dependencies

unittest2 does not build by now. Why do we use a package that had its
last release 8 years ago? FYI @tristan0x

matz-e committed:
deploy: typo in config generation w.r.t. environments

matz-e committed:
deploy: build neuron with nvhpc again

matz-e committed:
py-numpy: backport from upstream, pin setuptools as needed

matz-e committed:
deploy: temporarily disable neuron%nvhpc again

matz-e committed:
deploy: pin numpy

matz-e committed:
py-scipy: backport from upstream

heerener committed:
random123: patch became obsolete

matz-e committed:
Merge remote-tracking branch 'bbp/develop' into up-the-stream

matz-e committed:
deploy: style fixes

matz-e committed:
deploy: add older 1api™ for debugging

heerener committed:
py-ecdsa, py-python-jose: patches are obsolete

heerener committed:
py-python-jose,py-ecdsa,random123: when obsoleting patches, also get the upstream version of the package. Duh.

matz-e committed:
deploy: more numpy pain

matz-e committed:
py-neurodamus: pin numpy version

heerener committed:
py-dbf, py-dask-mpi: patches have been upstreamed

New dependencies have been added for newer versions

heerener committed:
py-elasticsearch-dsl: upstreamed

heerener committed:
py-jupyter-server-proxy: fix dependency condition

matz-e committed:
Merge branch 'develop' into up-the-stream
